File: 백준/Silver/9095. 1， 2， 3 더하기/1， 2， 3 더하기.swift
var dp = [Int](repeating: 0, count: 11)

dp[1] = 1
dp[2] = 2
dp[3] = 4

for i in 4...10 {
    dp[i] = dp[i - 3] + dp[i - 2] + dp[i - 1]
}

let t = Int(readLine()!)!

for _ in 0..<t {
    let n = Int(readLine()!)!
    print(dp[n])
}

File: 백준/Silver/9095. 1， 2， 3 더하기/1， 2， 3 더하기.py
import sys
input = sys.stdin.readline
print = sys.stdout.write

T = int(input())

dp = [0 for i in range(12)]
dp[1] = 1
dp[2] = 2
dp[3] = 4

for i in range(4, 11):
    dp[i] = dp[i - 1] + dp[i - 2] + dp[i - 3]

for _ in range(T):
    n = int(input())
    print(f"{dp[n]}\n")